driver-core/src/main/com/mongodb/internal/connection/SaslAuthenticator.java

```diff
@@ -176,6 +176,7 @@ private void sendSaslContinueAsync(final BsonInt32 conversationId, final byte[]
 
     private BsonDocument createSaslStartCommandDocument(final byte[] outToken) {
         return new BsonDocument("saslStart", new BsonInt32(1)).append("mechanism", new BsonString(getMechanismName()))
+                .append("options", new BsonDocument("skipEmptyExchange", new BsonInt32(1)))
```

Comment:
The spec calls for this option to be added only for SCRAM mechanism, not all SASL mechanisms.  To do that, you'll have to add to the SaslAuthenticator abstraction to allow for different SASL mechs to have different (or no) options

Reply:
I have made this method `protected` and overload it in ScramShaAuthenticator.